File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <math.h>


int main()
{
    double change = 0.0;
    
    do{
 
        printf("O hai! How much change is owed?\n");
        change = get_float();
        
    }while(change < 0);
    
    change = change * 100;
    change = round(change);
    int rounded = (int)change;
    int returned_coins = 0;
    
    while(rounded >= 25)
    {
        returned_coins++;
        rounded = rounded - 25;
    }
    
    while(rounded >= 10)
    {
        returned_coins++;
        rounded = rounded - 10;
    }
    
    while(rounded >= 5)
    {
        returned_coins++;
        rounded = rounded - 5;
    }
    
    while(rounded >= 1)
    {
        returned_coins++;
        rounded = rounded - 1;
    }
    
    printf("%i\n", returned_coins);
    
}

File: pset8/greedy.py

def main():
    
    while True:
        print("O hai! How much change is owed?")
        cash = float(input())
        if cash > 0:
            break
    
    # Convert dollars to cents and round
    cash *= 100
    cash = round(cash)
    change = int(cash)
    
    
    # Initialize amount of return coins
    coins = 0
    
    while change >= 25:
        coins += 1
        change -= 25
    while change >= 10:
        coins += 1
        change -= 10
    while change >= 5:
        coins += 1
        change -= 5
    while change >= 1:
        coins += 1
        change -= 1
        
    if change < 1:
        print(coins)
    
    
if __name__ == "__main__":
    main()